presto-main/src/main/java/com/facebook/presto/server/StageResource.java

```diff
@@ -16,14 +16,17 @@
 import com.facebook.presto.execution.QueryManager;
 import com.facebook.presto.execution.StageId;
 
+import javax.annotation.security.RolesAllowed;
 import javax.inject.Inject;
 import javax.ws.rs.DELETE;
 import javax.ws.rs.Path;
 import javax.ws.rs.PathParam;
 
+import static com.facebook.presto.server.security.RoleType.ADMIN;
 import static java.util.Objects.requireNonNull;
 
 @Path("/v1/stage")
+@RolesAllowed(ADMIN)
```

Comment:
This is used in `QueryResult` to allow for users to cancel their query - we should make this `USER`